homeassistant/components/matter/climate.py

```diff
@@ -143,46 +177,84 @@ def target_temperature(self) -> float | None:
                 return self._get_temperature_in_degrees(
                     clusters.Thermostat.Attributes.OccupiedHeatingSetpoint
                 )
-            case HVACMode.AUTO:
-                # When the system mode is set to "auto," there is no target temperature; instead, there is a target temperature low and high.
-                return None
             case _:
                 return None
 
-    async def async_set_temperature(self, **kwargs: Any) -> None:
-        """Set new target temperature."""
-        target_temp: float | None = kwargs.get(ATTR_TEMPERATURE)
+    @property
+    def target_temperature_high(self) -> float | None:
+        """Return the highbound target temperature we try to reach."""
+        if self.hvac_mode == HVACMode.HEAT_COOL:
+            return self._get_temperature_in_degrees(
+                clusters.Thermostat.Attributes.OccupiedCoolingSetpoint
+            )
+
+    @property
+    def target_temperature_low(self) -> float | None:
+        """Return the lowbound target temperature we try to reach."""
+        if self.hvac_mode == HVACMode.HEAT_COOL:
+            return self._get_temperature_in_degrees(
+                clusters.Thermostat.Attributes.OccupiedHeatingSetpoint
+            )
+
+    @staticmethod
+    def create_optional_setpoint_command(
+        mode: clusters.Thermostat.Enums.SetpointAdjustMode,
+        target_temp: float | None,
+        current_target_temp: float | None,
+    ) -> clusters.Thermostat.Commands.SetpointRaiseLower | None:
+        """Create a setpoint command if the target temperature is different from the current one."""
+        if target_temp is None or current_target_temp is None:
+            return None
 
-        if target_temp is None or self.target_temperature is None:
-            return
+        temp_diff = int((target_temp - current_target_temp) * 10)
 
-        temp_diff = int((target_temp - self.target_temperature) * 10)
+        if temp_diff == 0:
+            return None
 
+        return clusters.Thermostat.Commands.SetpointRaiseLower(
+            mode,
+            temp_diff,
+        )
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new target temperature."""
         match self.hvac_mode:
             case HVACMode.HEAT:
-                command = clusters.Thermostat.Commands.SetpointRaiseLower(
+                command = self.create_optional_setpoint_command(
                     clusters.Thermostat.Enums.SetpointAdjustMode.kHeat,
-                    temp_diff,
+                    kwargs.get(ATTR_TEMPERATURE),
```

Comment:
I think we should check that there is a target temperature parameter provided. It's optional. If it's missing we should raise `ValueError` with a message for this case. Same for the other cases below. If an expected parameter is missing we should raise early.